assets/js-maps.js

```diff
@@ -1,75 +1,116 @@
+      return await response.json()
+    } catch (error) {
+      console.error('Error fetching location data:', error);
+      return [];
     }
+  }
 
-    this.classes = {
-      error: "map__error",
-      noImage: "no-image"
+  const getIconDimensions = (iconSelector = config.selector.icon) => {
+    // Check cache first for better performance
+    if (cache.iconDimensions.has(iconSelector)) {
+      return cache.iconDimensions.get(iconSelector);
     }
 
-    this.addressErrorMessage = `we can't find this address, please check it again`;
-    this.linkArr = [];
-    this.zoom = 18;
-  }
+    const icon = document.querySelector(iconSelector);
+    if (!icon) return [40, 55] // Default dimensions
 
-  init() {
-    if (!this.block) return false;
+    const dimensions = $.getDimensions(icon);
+    const result = dimensions.width && dimensions.height
+      ? [dimensions.width, dimensions.height]
+      : [40, 55];
 
-    this.elements();
-    this.events();
-  }
+    cache.iconDimensions.set(iconSelector, result); // Cache the result for future use
 
-  elements() {
-    this.maps = this.block.querySelectorAll(this.selector.map);
-    this.links = this.block.querySelectorAll(this.selector.link);
+    return result;
   }
 
-  events() {
-    this.createMap();
-  }
+  const createMap = async (mapElement, address) => {
+    if (!mapElement || !address) return;
 
-  createMap() {
-    if (!this.maps.length) return false;
+    if (mapElement.getAttribute(config.attribute.processed) === 'true') return;
 
-    this.maps.forEach(map => {
-      const id = map.getAttribute(this.attr.id),
-            address = map.getAttribute(this.attr.address),
-            url = `https://nominatim.openstreetmap.org/search?q=${encodeURIComponent(address)}&format=json`;
+    const mapId = mapElement.getAttribute(config.attribute.id),
+          locationData = await fetchLocationData(address);
 
-      const getData = async () => {
-        try {
-          const res = await fetch(url);
-          return await res.json();
-        } catch (error) {
-          console.error(error);
-        }
-      };
+    if (!locationData || !locationData.length) {
+      displayErrorMessage(mapElement, address);
+      return;
+    }
+
+    renderMap(mapId, locationData);
+    updateMapLinks(locationData);
+
+    mapElement.setAttribute(config.attribute.processed, 'true'); // Mark as processed to prevent duplicate processing
+  }
 
-      const renderMap = (obj) => {
-        const icon = document.querySelector(this.selector.icon),
-              iconWidth = icon?.getBoundingClientRect().width,
-              iconHeight = icon?.getBoundingClientRect().height,
-              iconSize = iconWidth && iconHeight ? [iconWidth, iconHeight] : [40, 55],
-              lon = obj[0]?.lon,
-              lat = obj[0]?.lat;
+  const renderMap = (targetId, locationData) => {
+    const [lon, lat] = [locationData[0].lon, locationData[0].lat];
+    const iconDimensions = getIconDimensions(),
```

Comment:
pass here `config.selector.icon`? The function becomes dumber if no default variable assigned `iconSelector = config.selector.icon`?